server/util.go

```diff
@@ -129,7 +130,25 @@ func fromContext[T any](r *http.Request, key contextKey) (T, error) {
 func maybeDecodeHeader(header string) string {
 	decoded, err := mimeDecoder.DecodeHeader(header)
 	if err != nil {
-		return header
+		return cloudflarePriorityIgnore(header)
 	}
-	return decoded
+	return cloudflarePriorityIgnore(decoded)
```

Comment:
This function is used not just by the priority, but also by other headers. So one could not use a notification title `u=hello` anymore. Why not keep this logic only in the parsePrio function like it was before?

Reply:
Sorry sorry sorry 🤯
Sorry for the error, I was so focussed trying to solve specifically the problem with the CF priority header that I didn't think about the possibility of another header having a value starting with `u=`, even knowing this function is used with with all headers.

> Why not keep this logic only in the parsePrio function like it was before? 

You mean this?
https://github.com/binwiederhier/ntfy/blob/2305ebca244fb60aa784a3b79270ae6f33b8ee48/util/util.go#L163-L170

Well, if you're referring to that part of the code, it's because at that point of execution, we have already parsed the headers and query parameters. If we find any priority header (priority, x-priority, prio, p), the others will not even be read, including the query parameters.

For instance, if a user is using the web interface to send a notification with priority 5, the request will include a query string like this:
`?title=titulodeprueba&priority=5`

If we send the request directly to the server, all will be ok, it will work, but if the server is behind CF, it will (in the majority of cases) have the `priority:"u=*"` header, causing the priority set in the query parameter to be ignored.

Nevertheless, I am fixing this issue in my next commit.
Thank you for all your comments.
😉